src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -545,12 +546,20 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
         Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest, request)
         for request in entry_point_requests
     )
+    entry_points = setup_kwargs.setdefault("entry_points", {})
```

Comment:
This will unconditionally add `entry_points` to the generated setup.py file. Maybe not such a big deal.. ?
I did leave the console_scripts in the for-loop on purpose, as we don't need that outside the for-loop.

Reply:
It will probably be friendly to future debugging sessions to leave it out if there are none.

How about just `entry_points = defaultdict(list)` to simplify the for loops a bit and then one line of post-processing that says:
```python
if entry_points:
  setup_kwargs["entry_points"] = entry_points
```

```